internal/search/query/parser.go

```diff
@@ -86,18 +88,25 @@ const (
 	RPAREN keyword = ")"
 )
 
-func isSpace(c byte) bool {
-	return (c == ' ') || (c == '\n') || (c == '\r') || (c == '\t')
+func isSpace(buf []byte) bool {
+	r, _ := utf8.DecodeRune(buf)
+	return unicode.IsSpace(r)
 }
 
-// skipSpace returns the number of spaces skipped from the beginning of a buffer buf.
+// skipSpace returns the number of spaces, corresponding to runes, skipped from the beginning of a buffer buf.
 func skipSpace(buf []byte) int {
-	for i, c := range buf {
-		if !isSpace(c) {
-			return i
+	pos := 0
+	for {
+		if pos > len(buf) {
```

Comment:
```suggestion
		if pos >= len(buf) {
```

This code is a bit tricky. Usually this sort of code is clearer if you update buf to point to the subslice.

eg something like
```go
count := 0
for len(buf) > 0 {
  r, advance := utf8.DecodeRune(buf)
  if !unicode.IsSpace(r) {
    break
  }
  count += advance
  buf = buf[advance:]
}
return count
```